File: pset-2-readability/readability.c
#include <cs50.h>
#include <stdio.h>

#include <string.h>
#include <ctype.h>
#include <math.h>

//count functions
int count_letters(string s);
int count_words(string s);
int count_sentences(string s);

//simple average function - better conversion to double
double avg(int x, int y);

int main(void)
{
    //get input text
    string text = get_string("Text: ");

    //count and store
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    //enumerate average for L and S
    double avg_L = avg(letters, words);
    double avg_S = avg(sentences, words);

    //assign a grade
    double index = (0.0588 * avg_L) - (0.296 * avg_S) - 15.8;
    int grade = round(index);

    //output
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

}

//letters
int count_letters(string s)
{
    int result = 0;
    for (int i = 0; i < strlen(s); i++)
    {
        if (isalpha(s[i]))
        {
            result++;
        }
    }
    return result;
}

//words
int count_words(string s)
{
    int result = 1;
    char before = ' ';
    char after = ' ';
    for (int i = 0; i < strlen(s) - 1; i++)
    {
        after = s[i + 1];
        if (s[i] == ' ' && (isalpha(after) || isalpha(before)))
        {
            result++;
        }
        before = s[i];
    }
    return result;
}

//sentences
int count_sentences(string s)
{
    int result = 0;
    for (int i = 0; i < strlen(s); i++)
    {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            result++;
        }
    }
    return result;
}

//average
double avg(int x, int y)
{
    double result = x * 100;
    result /= y;
    return result;
}

File: pset-6-sentimental-readability/readability.py
# Main
def main():
    text = input("Text: ")

    # Count everything up
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Enumerate average
    avg_L = avg(letters, words)
    avg_S = avg(sentences, words)

    # Enumerate grade
    index = (0.0588 * avg_L) - (0.296 * avg_S) - 15.8
    grade = round(index)

    # Output
    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print("Grade " + str(grade))


# Count letters in text
def count_letters(s):
    result = 0
    for letter in s:
        if letter.isalpha():
            result += 1
    return result


# Count words in text
def count_words(s):
    result = 1
    before = ' '
    after = ' '
    for i in range(0, len(s)-1):
        after = s[i + 1]
        if (s[i] == ' ' and (after.isalpha() or before.isalpha())):
            result += 1
        before = s[i]
    return result


# Count sentences in text
def count_sentences(s):
    result = 0
    for i in range(0, len(s)):
        if (s[i] == '.' or s[i] == '!' or s[i] == '?'):
            result += 1
    return result


# Simple average function
def avg(x, y):
    result = x * 100
    result /= y
    return result


if __name__ == "__main__":
    main()